src/main/java/com/webapp/TextBook/sharedFiles/ValidationBindingHelper.java

```diff
@@ -0,0 +1,181 @@
+package com.webapp.TextBook.sharedFiles;
+
+import com.sun.istack.Nullable;
+import com.webapp.TextBook.validation.Shared.SharedValidationState;
+import org.javatuples.Pair;
+import org.springframework.boot.configurationprocessor.json.JSONArray;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.boot.configurationprocessor.json.JSONObject;
+import org.springframework.boot.context.properties.bind.validation.BindValidationException;
+import org.springframework.ui.ModelMap;
+import org.springframework.validation.BindException;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.ObjectError;
+
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ValidationBindingHelper {
+
+    private static final String failedValidationStatusMessage = "Input missing or invalid";
+    private static final String INTERNAL_VALIDATION_ERROR = "Internal Error: please contact IT.";
+
+    // creates local final strings for error and general errors key
+    private static final String ERRORS_KEY = "Errors";
+    private static final String GENERAL_ERRORS_KEY = "GeneralError";
+
+    public static Pair<Boolean,JSONObject> validationBindingHandler(@NotNull final BindingResult result, @NotNull final ModelMap map) throws JSONException {
+        JSONObject data = new JSONObject();
```

Comment:
more comments needed: what do there local variables mean (why do they exist in the context of this method)